src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php

```diff
@@ -94,6 +97,15 @@ public static function register(ContainerBuilder $container, array $refMap)
             $container->setDefinition($id, $locator);
         }
 
+        if (null !== $callerId = 2 < func_num_args() ? func_get_arg(2) : null) {
+            $locatorId = $id;
+            $container->register($id .= '.'.$callerId, ServiceLocator::class)
```

Comment:
I know you hate it when I suggest this, but a comment above this explaining what's going on would help. I can't locate where and why registering this service is significant.